singer_sdk/sinks/core.py

```diff
@@ -410,3 +431,31 @@ def clean_up(self) -> None:
         should not be relied on, it's recommended to use a uuid as well.
         """
         pass
+
+    def process_batch_file(
+        self,
+        encoding: BaseBatchFileEncoding,
+        storage: StorageTarget,
+        path: str,
+    ) -> None:
```

Comment:
As an overrideable interface, I do think this needs to be pluralized. (Relates to my comment in the other thread.)

For platforms like Redshift, Snowflake, etc., is a significantly suboptimal load path if we pass a sequence of calls to `process_batch_file()`, versus using `process_batch_files()` as the public interface and anyone who overrides the method handles this as "1 or more" files.

There's a significant logical distinction between "10 batches of 1 file each" versus "2 batches of 5 files each". 

--------- 

Redshifts docs are very clear on the importance of loading multiple files in one step:

https://docs.aws.amazon.com/redshift/latest/dg/c_best-practices-single-copy-command.html

> ## Use a single COPY command to load from multiple files
>
> Amazon Redshift can automatically load in parallel from multiple compressed data files.
> 
> _However, if you use multiple concurrent COPY commands to load one table from multiple files, Amazon Redshift is forced to perform a serialized load._ **_This type of load is much slower and requires a VACUUM process at the end if the table has a sort column defined._** 

Something like this?:

```suggestion
    def process_batch_files(
        self,
        encoding: BaseBatchFileEncoding,
        storage: StorageTarget,
        paths: List[str],
    ) -> None:
```

Reply:
@aaronsteers that makes sense. The default `process_batch_files` implementation can still iterate over the files and other Sink subclasses can do their own 👌.

Reply:
Changes pushed